networks/hamiltonian_net.py

```diff
@@ -0,0 +1,116 @@
+            n_filters (list): List with number of filters for each of the hidden layers.
+            kernel_sizes (list): List with kernel sizes for each convolutional layer.
+            strides (list): List with strides for each convolutional layer.
+            act_func (torch.nn.Module): The activation function to apply after each layer.
+            dtype (torch.dtype): Type of the weights.
+        """
+        super().__init__()
+        if all(var is None for var in (hidden_conv_layers, n_filters, kernel_sizes, strides)):
+            hidden_conv_layers = HamiltonianNet.DEFAULT_PARAMS['hidden_conv_layers']
+            n_filters = HamiltonianNet.DEFAULT_PARAMS['n_filters']
+            kernel_sizes = HamiltonianNet.DEFAULT_PARAMS['kernel_sizes']
+            strides = HamiltonianNet.DEFAULT_PARAMS['strides']
+        elif all(var is not None for var in (hidden_conv_layers, n_filters, kernel_sizes, strides)):
+            # If no Nones, check consistency
+            assert len(n_filters) == hidden_conv_layers + 2,\
+                'n_filters must be a list of length hidden_conv_layers + 2 ' \
+                '(' + str(hidden_conv_layers + 2) + ' in this case).'
+            assert len(kernel_sizes) == hidden_conv_layers + 2 and \
+                   len(strides) == hidden_conv_layers + 2, \
+                   'kernel_sizes and strides must be lists with values foreach layer in the ' \
+                   'network (' + str(hidden_conv_layers + 2) + ' in this case).'
+        else:
+            raise ValueError('Args hidden_conv_layers, n_filters, kernel_sizes, and strides'
+                             'can only be either all None, or all defined by the user.')
+        in_channels = in_shape[0] * 2
+        paddings = [int(k/2) for k in kernel_sizes]
+        self.in_conv = nn.Conv2d(
+            in_channels=in_channels,
+            out_channels=n_filters[0],
+            kernel_size=kernel_sizes[0],
+            padding=paddings[0])
+        out_size = int((in_shape[1] - kernel_sizes[0] + 2 * paddings[0])) / strides[0] + 1
+        self.hidden_layers = nn.ModuleList(
+            modules=[
+                nn.Conv2d(
+                    in_channels=n_filters[i],
+                    out_channels=n_filters[i + 1],
+                    kernel_size=kernel_sizes[i + 1],
+                    padding=paddings[i + 1],
+                    stride=strides[i + 1]
+                )
+                for i in range(hidden_conv_layers)
+            ]
+        )
+        for i in range(hidden_conv_layers):
+            out_size = int((out_size - kernel_sizes[i] + 2 * paddings[i]) / strides[i]) + 1
+        self.out_conv = nn.Conv2d(
+            in_channels=n_filters[-1],
+            out_channels=n_filters[-1],
+            kernel_size=kernel_sizes[-1],
+            padding=paddings[-1]
+        )
+        out_size = int((out_size - kernel_sizes[-1] + 2 * paddings[-1]) / strides[-1]) + 1
+        self.n_flat = out_size ** 2 * n_filters[-1]
```

Comment:
maybe put parenthesis just in case

Reply:
You mean `self.n_flat = (out_size ** 2) * n_filters` ?

Reply:
yes

Reply:
Done in 91a0f67